src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -415,10 +475,22 @@ private void TrackNullableStateForAssignment(BoundNode node, Symbol assignmentTa
                             valueIsNotNull;
                     }
 
-                    if (slot > 0 && targetType.TypeSymbol.IsAnonymousType && targetType.TypeSymbol.IsClassType() &&
-                        (value == null || targetType.TypeSymbol == value.Type))
+                    if (slot > 0)
                     {
-                        InheritNullableStateOfAnonymousTypeInstance(targetType.TypeSymbol, slot, GetValueSlotForAssignment(value), isByRefTarget);
+                        InheritDefaultState(slot);
+
+                        // PROTOTYPE(NullableReferenceTypes): We're copying fields of anonymous type assignments,
+                        // even if the RHS is an alias. Is that the expected behavior or should we ignore aliases in this case?
+                        if (value != null &&
```

Comment:
>value != null && [](start = 28, length = 16)

Addition of this condition doesn't feel right.

Reply:
We're only copying fields that were set in `value`.